git-sub

```diff
@@ -53,25 +74,15 @@ function subClone() {
 
 	cd $sub
 
-	# init
-	git init
-
-	echo "Add remote $remoteName: $remote"
-	git remote add $remoteName $remote
-
-	# fetch
-	echo "Fetching..."
-	git fetch $remoteName
-
-	# ref
-	if [ "$ref" != "" ]; then
-		echo "Checkout $ref"
-		git checkout -f $ref
+	if [ -z "$ref" ]; then
+		git clone $remote . --quiet
+		ref=$(git symbolic-ref --short HEAD) 
 	else
-		# Use default branch as stored ref
-		ref=$(git branch --color=never 2> /dev/null | sed -e '/^[^*]/d' -e 's/* \(.*\)/\1/')
+		git clone $remote . --branch $ref --quiet
 	fi
```

Comment:
Why not just cloning and then checking ref for a later checkout?
```bash
	git clone $remote . --quiet
	if [ -z "$ref" ]; then
	    ref=$(git symbolic-ref --short HEAD)
	else
	    git checkout $ref
	fi
```
This way the clone command is the same and it may improve readibility

Reply:
Nice suggestion. I'll simplify the cloning.